maas/client/viscera/machines.py

```diff
@@ -190,18 +208,107 @@ class Machine(Object, metaclass=MachineType):
         if comment is not None:
             params["comment"] = comment
         data = await self._handler.deploy(**params)
-        return type(self)(data)
+        if not wait:
+            return type(self)(data)
+        else:
+            # Wait for the machine to be fully deployed
+            machine = type(self)(data)
+            while machine.status == NodeStatus.DEPLOYING:
+                await asyncio.sleep(wait_interval)
+                data = await self._handler.read(system_id=self.system_id)
+                machine = type(self)(data)
+            if machine.status == NodeStatus.FAILED_DEPLOYMENT:
+                msg = "{system_id} failed to Deploy.".format(
+                    system_id=machine.system_id
+                )
+                raise FailedDeployment(msg)
```

Comment:
I agree with the principle here, but just wondering if there's a better, more generic way to do it? Seems like something we'd want to do generally is allow the user to access the object the exception applies to, from that exception. Not sure doing it piecemeal per method is the right way...